File: Java/Programmers/Level1/숫자 문자열과 영단어.java
//https://programmers.co.kr/learn/courses/30/lessons/81301
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = 0;

        Map<String, String> map = new HashMap<>();

        map.put("zero", "0");
        map.put("one", "1");
        map.put("two", "2");
        map.put("three", "3");
        map.put("four", "4");
        map.put("five", "5");
        map.put("six", "6");
        map.put("seven", "7");
        map.put("eight", "8");
        map.put("nine", "9");


        Iterator<String> keys = map.keySet().iterator();
        while(keys.hasNext()) {
            String key = keys.next();
            s = s.replace(key, map.get(key));
        }

        return Integer.parseInt(s);
    }
}

File: JavaScript/Programmers/Level1/숫자 문자열과 영단어.js
function solution(s) {
  for(let i=0; i<map.length; i++) {
    while(s.includes(map[i]['key'])) {
      s = s.replace(map[i]['key'], map[i]['value']);
      if(!s.includes(map[i]['key']))
        break;
    }
  }
  return Number(s);
}


let map = [
  {"key" : "zero", "value" : 0},
  {"key" : "one", "value" : 1},
  {"key" : "two", "value" : 2},
  {"key" : "three", "value" : 3},
  {"key" : "four", "value" : 4},
  {"key" : "five", "value" : 5},
  {"key" : "six", "value" : 6},
  {"key" : "seven", "value" : 7},
  {"key" : "eight", "value" : 8},
  {"key" : "nine", "value" : 9},
]

let s = "1zerotwozero3"
console.log(solution(s));

/*
function solution(s) {
    let numbers = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
    var answer = s;

    for(let i=0; i< numbers.length; i++) {
        let arr = answer.split(numbers[i]);
        answer = arr.join(i);
    }

    return Number(answer);
}
*/
